### PR TITLE
Flot annotations2

### PR DESCRIPTION
**Pull request by andrewdove1**
Updated based on wireframes
Added method hitTest = will be used to modify annotation when mouse if over it
Added method getBounds = will be used to create html elements superimposed on the annotation
defaultxaxis & defaultyaxis - absolute coords can be relative to any axis per Brian

### CODE REVIEW

File: docs/NationalInstruments/flot-annotations.md [hunk 1]
```diff
@@ -85,6 +91,15 @@ The plugin adds some public methods to the plot:
     remove the specified annotation from the plot. annotationToRemove is an annotation
     reference to one of the annotations obtained with getAnnotations()
 
+* plot.hitTest(x, y)
```
Comment (brianehenry) [inline docs/NationalInstruments/flot-annotations.md]: Discussed this in person, but these method names should be more specific.

File: tests/flot.annotations.Test.js [hunk 2]
```diff
@@ -124,6 +127,49 @@ describe('Flot annotations', function () {
...
+            x2: 7,
+            y2: 12
+        });
+
+        var expectedX1 = pos1.left;
+        var expectedY1 = pos1.top;
+        var spy1 = spyOn(ctx, 'fillRect').and.callThrough();
+        plot.triggerRedrawOverlay();
+        jasmine.clock().tick(20);
+        expect(spy1).not.toHaveBeenCalledWith(expectedX1, expectedY1, jasmine.any(Number), jasmine.any(Number));
```
Comment (atmgrifter00) [inline tests/flot.annotations.Test.js]: I don't think this test clearly tests what we are trying to describe here. I believe a proper test would explicitly have two sets of axes with different ranges (which is implicit in this test...driven by the data autoscaling the axes), and then two annotations assigned to a different axis-pair, such that (for ease of testing) they overlapped exactly in the plot area. The test would then make sure that we called the same 'fillRect' method twice with the same values despite them being positioned to two different sets of domain values.

File: source/NationalInstruments/jquery.flot.annotations.js [hunk 3]
```diff
@@ -153,15 +197,16 @@ THE SOFTWARE.
...
             if (!annotation.showArrow) {
                 return offset;
             }
 
             let x = annotation.x;
             let y = annotation.y;
-            let xaxis = plot.getXAxes()[0];
-            let yaxis = plot.getYAxes()[0];
+            let xaxis = plot.getXAxes()[zeroBasedIndex];
+            let yaxis = plot.getYAxes()[zeroBasedIndex];
```
Comment (brianehenry) [inline source/NationalInstruments/jquery.flot.annotations.js]: Should be using different config values for each axis.

File: tests/flot.annotations.Test.js [hunk 4]
```diff
@@ -203,4 +250,80 @@ describe('Flot annotations', function () {
...
+            y1: 9
+        });
+
+        var x1 = pos1.left / plot.width();
+        var y1 = pos1.top / plot.height();
+        let annotations = plot.hitTest(x1, y1);
+        expect(annotations.length).toEqual(1);
+        expect(annotations[0]).toEqual(0);
+    });
+    fit('should return the bounds of an annotation using the public api', function () {
```
Comment (atmgrifter00) [inline tests/flot.annotations.Test.js]: change to 'it'

File: tests/flot.annotations.Test.js [hunk 5]
```diff
@@ -203,4 +250,80 @@ describe('Flot annotations', function () {
...
+        var pos1 = plot.p2c({
+            x1: 4,
+            y1: 9
+        });
+
+        var x1 = pos1.left / plot.width();
+        var y1 = pos1.top / plot.height();
+        let annotations = plot.hitTest(x1, y1);
+        expect(annotations.length).toEqual(1);
+        expect(annotations[0]).toEqual(0);
```
Comment (brianehenry) [inline tests/flot.annotations.Test.js]: What is this supposed to be asserting?

File: tests/flot.annotations.Test.js (hunk 4 above)
Comment (brianehenry) [inline tests/flot.annotations.Test.js]: Remove `fit`

File: tests/flot.annotations.Test.js (hunk 5 above)
Comment (andrewdove1) [reply tests/flot.annotations.Test.js]: added a comment

File: docs/NationalInstruments/flot-annotations.md [hunk 6]
```diff
@@ -107,11 +122,12 @@ var myFlot = $.plot( $("#graph"), ...,
                 borderColor: '#FF0000',
```
[the remark this answers is not in the document]
Comment (brianehenry) [reply docs/NationalInstruments/flot-annotations.md]: You'll also need to update your examples for the API changes. I'm okay with you fixing these in a future PR.

**Pull request merged**